File: fix_fonts.py
import re

with open('src/app/learnmore/auto-conversion-2d-to-3d/page.tsx', 'r', encoding='utf-8') as f:
    lines = f.readlines()

new_lines = []
for i, line in enumerate(lines):
    # Only replace text-lime if it's in a span or div (not an icon like <Check className="text-lime" />)
    if 'group-hover:text-lime' in line:
        line = line.replace('group-hover:text-lime', 'group-hover:text-zinc-600 dark:group-hover:text-zinc-300')
    
    if '<span' in line or '<div' in line:
        # Check if it contains text-lime but NOT if it's just a wrapper around an icon
        if 'text-lime' in line and not '<Check' in line and not '<Sparkles' in line and not '<Cpu' in line and not '<Cube' in line and not '<Play' in line and not 'text-lime shrink-0' in line and not 'rounded-full bg-lime/10 border border-lime/30 flex items-center justify-center text-lime' in line and not 'w-8 h-8 rounded-lg bg-lime/10 border border-lime/20 flex items-center justify-center text-lime' in line:
            line = line.replace('text-lime', 'text-zinc-900 dark:text-white')

    new_lines.append(line)

with open('src/app/learnmore/auto-conversion-2d-to-3d/page.tsx', 'w', encoding='utf-8') as f:
    f.writelines(new_lines)

print("Done replacing fonts.")

File: fix_fonts.js
const fs = require('fs');

const path = 'src/app/learnmore/auto-conversion-2d-to-3d/page.tsx';
const content = fs.readFileSync(path, 'utf8');
const lines = content.split('\n');

const newLines = lines.map(line => {
    let replacedLine = line;

    // Replace hover:text-lime
    if (replacedLine.includes('group-hover:text-lime')) {
        replacedLine = replacedLine.replace(/group-hover:text-lime/g, 'group-hover:text-zinc-600 dark:group-hover:text-zinc-300');
    }

    // Replace text-lime in text tags (span, div)
    if (replacedLine.includes('<span') || replacedLine.includes('<div') || replacedLine.includes('{activeScene === 3 && <span')) {
        // Exclude lines that only use text-lime for icons
        if (
            replacedLine.includes('text-lime') &&
            !replacedLine.includes('<Check') &&
            !replacedLine.includes('<Sparkles') &&
            !replacedLine.includes('<Cpu') &&
            !replacedLine.includes('<Cube') &&
            !replacedLine.includes('<Play') &&
            !replacedLine.includes('text-lime shrink-0') &&
            !replacedLine.includes('rounded-full bg-lime/10 border border-lime/30 flex items-center justify-center text-lime') &&
            !replacedLine.includes('w-8 h-8 rounded-lg bg-lime/10 border border-lime/20 flex items-center justify-center text-lime') &&
            !replacedLine.includes('w-16 h-16 rounded-full bg-lime/10 border border-lime/30 flex items-center justify-center text-lime') &&
            !replacedLine.includes('w-12 h-12 rounded-full bg-lime/10 border border-lime/30 flex items-center justify-center text-lime') &&
            !replacedLine.includes('w-[85%] h-[85%] relative z-10 text-lime') &&
            !replacedLine.includes('w-[80%] h-[80%] relative z-10 text-lime') &&
            !replacedLine.includes('className="w-full h-full text-lime p-2 sm:p-6"') &&
            !replacedLine.includes('w-4 h-4 text-lime') &&
            !replacedLine.includes('w-12 h-12 text-lime mb-2 animate-bounce')
        ) {
            replacedLine = replacedLine.replace(/text-lime/g, 'text-zinc-900 dark:text-white');
        }
    }

    return replacedLine;
});

fs.writeFileSync(path, newLines.join('\n'), 'utf8');
console.log('Fonts updated successfully.');
